File: app/src/main/java/com/nytsample/android/view/NewsDetailsFragment.java
package com.nytsample.android.view;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.nytsample.android.R;
import com.nytsample.android.data.model.NewsFeedItemModel;
import com.nytsample.android.utils.ImageViewUtils;
import com.nytsample.android.view.def.NewsDetailsView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class NewsDetailsFragment extends Fragment implements NewsDetailsView {

    private static final String ARG_NEWS_FEED_ITEM = "news_feed_item";

    @BindView(R.id.imageViewPhoto)
    ImageView imageViewPhoto;
    @BindView(R.id.textViewTitle)
    TextView textViewTitle;
    @BindView(R.id.textViewByLine)
    TextView textViewByLine;
    @BindView(R.id.textViewDate)
    TextView textViewDate;

    public static NewsDetailsFragment newInstance(NewsFeedItemModel newsFeedItemModel) {
        NewsDetailsFragment fragment = new NewsDetailsFragment();
        Bundle bundle = new Bundle();
        bundle.putSerializable(ARG_NEWS_FEED_ITEM, newsFeedItemModel);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_news_details, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        NewsFeedItemModel newsFeedItemModel = getNewsFeedItemArg();
        if (newsFeedItemModel != null) {
            showDetails(newsFeedItemModel);
        } else {
            throw new IllegalStateException("Use NewsDetailsFragment.newInstance(NewsFeedItemModel newsFeedItemModel)");
        }
    }

    @Override
    public void showDetails(NewsFeedItemModel newsFeedItem) {
        ImageViewUtils.loadImage(imageViewPhoto, newsFeedItem.getPhotoUrl());
        textViewTitle.setText(newsFeedItem.title);
        textViewByLine.setText(newsFeedItem.byLine);
        textViewDate.setText(newsFeedItem.publishedDate);
    }

    private NewsFeedItemModel getNewsFeedItemArg() {
        if (getArguments() != null && getArguments().containsKey(ARG_NEWS_FEED_ITEM)) {
            return (NewsFeedItemModel) getArguments().getSerializable(ARG_NEWS_FEED_ITEM);
        }
        return null;
    }
}

File: app/src/main/java/com/nytsample/android/view/NewsDetailsFragment.kt
package com.nytsample.android.view

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.nytsample.android.R
import com.nytsample.android.data.model.NewsFeedItemModel
import com.nytsample.android.utils.load
import com.nytsample.android.view.def.NewsDetailsView
import kotlinx.android.synthetic.main.fragment_news_details.*

class NewsDetailsFragment : Fragment(), NewsDetailsView {

    companion object {
        const val ARG_NEWS_FEED_ITEM = "news_feed_item"

        fun newInstance(newsFeedItem: NewsFeedItemModel) = NewsDetailsFragment().apply {
            val args = Bundle()
            args.putSerializable(ARG_NEWS_FEED_ITEM, newsFeedItem)
            arguments = args
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_news_details, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)


        val newsFeedItemModel = getNewsFeedItemArg()
        if (newsFeedItemModel != null) {
            showDetails(newsFeedItemModel)
        } else {
            throw IllegalStateException("Use NewsDetailsFragment.newInstance(newsFeedItem: NewsFeedItemModel)")
        }
    }

    override fun showDetails(newsFeedItem: NewsFeedItemModel) {
        imageViewPhoto.load(newsFeedItem.getPhotoUrl())
        textViewTitle.text = newsFeedItem.title
        textViewByLine.text = newsFeedItem.byLine
        textViewDate.text = newsFeedItem.publishedDate
    }

    private fun getNewsFeedItemArg(): NewsFeedItemModel? {
        if (arguments != null && arguments!!.containsKey(ARG_NEWS_FEED_ITEM)) {
            return arguments?.getSerializable(ARG_NEWS_FEED_ITEM) as NewsFeedItemModel
        }
        return null
    }
}